hummingbot/hummingbot
# Feat/add creation timestamp to in flight orders

**Before submitting this PR, please make sure**:


**A description of the changes proposed in the pull request**:
We need to start registering the orders creation timestamp as a separate field. Currently orders take that information from the last component of the order id. But there is a requirement to change the order id generation method, to ensure its uniqueness, and the creation timestamp will not be part of the order id any more in the future.

This PR adds the creation timestamp in the inflight orders. It also changes the calculation of the orders age that was done by taking the creation timestamp from the client order id. Applies the same changes to the hanging orders.
Modifies the JSON serialization/deserialization of orders to include the creation timestamp.
Changes the market recorder to use the new timestamp when storing the creation time instead of using the timestamp when the event was received (the previous logic).


**Tests performed by the developer**:
Added unit tests to validate the serialization/deserialization and the market recorder changes.
Changed existing tests for hanging orders.


**Tests performed**:
- Create and run pure market making and avellaneda market making strategies on spot connectors
- Confirmed that hanging order execution and cancellation works on both strategies and no ghost hanging orders are created when restarting the strategy
- Create and run perpetual market making on perpetual connectors
- Create and run amm_arb strategy on protocol connectors
- Confirmed that filled order event on tested strategies works as expected
- Confirmed all issues found were addressed and fixed

**Tips for QA testing**:
Since the PR modified the in flight orders for all connectors it is required to check that all of them continue working correctly. Please also test running either Pure Market Making or Avellaneda strategy using hanging orders.


This PR comes from https://github.com/CoinAlpha/hummingbot/pull/55
